MigrationV3V4/Node/Skills/README.md

```diff
@@ -1,20 +1,64 @@
+        enableSkills: true,
+        authConfiguration: new skills.AuthenticationConfiguration([], allowedCallersClaimsValidator)
+    });
+    ```
 
-```
-MICROSOFT_APP_ID={V3_BOT_APP_ID}
-MICROSOFT_APP_PASSWORD={V3_BOT_PASSWORD}
-ROOT_BOT_APP_ID={V4_BOT_APP_ID}
-```
+2)	Create a Skills Validator, similar to `v3-booking-bot-skill`'s `allowedCallersClaimsValidator.js`. The `allowedCallers` array can restrict which appId's can access the skill. Set this to "*" to accept calls from any skill consumer, or set it to a comma-delimited list of the app IDs of the allowed consumers. In these samples, we're only authorizing the parent bot, via `process.env.ROOT_BOT_APP_ID` which is set in the `.env` file.
+3)	Add code to send the skill host and `endOfConversation` activity when the skill is finished and to properly handle `endOfConversation` from the skill consumer.
+
+(Note: these changes will not affect the bot's behavior as a normal v3 bot, and it can still be used that way, too.)
+
+## Prerequisites
+
+The minimum prerequisites to run this sample are:
+* Latest Node.js with NPM. Download it from [here](https://nodejs.org/en/download/).
+* The Bot Framework Emulator. To install the Bot Framework Emulator, download it from [here](https://emulator.botframework.com/). Please refer to [this documentation article](https://github.com/microsoft/botframework-emulator/wiki/Getting-Started) to know more about the Bot Framework Emulator.
+* **[Recommended]** Visual Studio Code for IntelliSense and debugging, download it from [here](https://code.visualstudio.com/) for free.
+
+## Key concepts in this sample
+
+The solution includes a parent bot (`v4-root-bot`) built using [v4.7.2](https://www.npmjs.com/package/botbuilder/v/4.7.2) and two V3 skill bots (v3-skill-bot and v3-booking-bot-skill) built using [v3.30.0](https://www.npmjs.com/package/botbuilder/v/3.30.0). This example demonstrates how a parent bot can post activities to skill bots and returns the skill responses to the user.
+
+- `v4-root-bot`: this project is based on [80.skills-simple-bot-to-bot](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/javascript_nodejs/80.skills-simple-bot-to-bot). It shows how to consume the two V3 skills and includes:
+  - A [RootBot](v4-root-bot/index.js) which calls the user's chosen skill and keeps the conversation active until the user says "end" or "stop". [RootBot](v4-root-bot/index.js) also keeps track of the conversation with the skill and handles the `endOfConversation` activity received from the skill to terminate the conversation.
+  - A simple [SkillConversationIdFactory](v4-root-bot/skillConversationIdFactory.js) based on an in memory JavaScript `object` that creates and maintains conversation IDs used to interact with a skill.
+  - A [SkillsConfiguration](v4-root-bot/skillsConfiguration.js) class that loads skill definitions from `.env`.
+  - A [SkillController](v4-root-bot/rootBot.js) that handles skill responses.
```

Comment:
[SkillController] might be a copy (maybe [RootBot] or something, and [Index] for the one above?)

Reply:
Ah, missed that one. good catch!